src/ipa_pc/mod.rs

```diff
@@ -533,22 +745,17 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
             let degree_bound = labeled_polynomial.degree_bound();
             let hiding_bound = labeled_polynomial.hiding_bound();
             let commitment = labeled_commitment.commitment();
+            let randomness = labeled_randomness.randomness();
 
-            combined_polynomial += (cur_challenge, polynomial);
-            combined_commitment_proj += &commitment.comm.mul(cur_challenge);
-
-            if hiding_bound.is_some() {
-                has_hiding = true;
-                combined_rand += &(cur_challenge * &randomness.rand);
-            }
-
-            cur_challenge = opening_challenges(opening_challenge_counter);
-            opening_challenge_counter += 1;
+            let p_len = polynomial.coeffs.len();
+            let segments_count = p_len / key_len + if p_len % key_len != 0 { 1 } else { 0 };
 
-            let has_degree_bound = degree_bound.is_some();
+            let degree_bound_len = degree_bound.and_then(|degree_bound| {
+                if (degree_bound + 1) % key_len != 0 { Some(degree_bound + 1) } else { None }
```

Comment:
We should add a comment:
If the `degree_bound` is a multiple of the `key_len` then there is no need to prove the degree bound polynomial identity.
However, one still needs to check if the multiple of the key_len, i.e. the number of segments, corresponds to the claimed degree bound.

Reply:
I also added a comment to the verifier

Reply:
Comments are added